runtime/vm/valueTypeHelpers.cpp

```diff
@@ -54,4 +54,29 @@ defaultValueWithUnflattenedFlattenables(J9VMThread *currentThread, J9Class *claz
                 }
         }
 }
+
+BOOLEAN
+valueTypeCapableAcmp(J9VMThread *currentThread, j9object_t lhs, j9object_t rhs)
+{
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+		bool acmpResult = false;
+		if (rhs == lhs) {
+			acmpResult = true;
+		} else if ((NULL == rhs) || (NULL == lhs)) {
```

Comment:
Maybe write this as an else followed by the if, which wraps the class fetch/call.